### ISSUE TITLE
How to use on ios

### ISSUE THREAD

**Issue by Yumega** [jonjomckay/fritter#55]
How to use on ios,Thanks

**Comment by jonjomckay** [comment]
I actually tried to get Fritter to run using my old Macbook yesterday - it seems to run fine in the iOS simulator with no code changes needed!

Next steps I guess would be to figure out how to publish the app. It might be possible to sideload the IPA like on Android, but I doubt it would be distributable...

I'll need to investigate more, unless someone else can help!

**Comment by jonjomckay** [comment]
I forgot to add, if you wanted to try it for yourself on iOS, I managed to get it to work by:

1. Installing Xcode
2. Installing Flutter, using the instructions on [flutter.dev](https://flutter.dev)
3. Cloning the Fritter repository
4. Running `flutter run` from inside the Fritter repository

**Comment by SuperSonicHub1** [comment]
Have you considered adding Fritter to the AltStore: https://altstore.io/

**Comment by jonjomckay** [comment]
@SuperSonicHub1 thanks for the link, that project looks pretty neat! I had a look around, but couldn't find any information about how to submit/configure a new app - do you know if it's possible?

**Comment by SuperSonicHub1** [comment]
You can't submit as of yet, it should be coming soon, but you *can* sideload IPAs: https://altstore.io/faq/

You could set up a system similar to the Android app where the iOS app phones home to see if there's a new GitHub release, and then redirect the user to download and sideload the IPA.

**Comment by salevdns** [comment]
I have build an unsigned ipa, that can be installed on iOS devices using AltStore (or any other sideloading method of your choice). It runs fine on my iPad (iPadOS 14).
If anyone wants to try it **and** trusts me you can get it here:
[https://mega.nz/file/YZ0ERbiQ](https://mega.nz/file/YZ0ERbiQ#hJHoFj39NbLb-c9dAe82-bMchS3UhK3cdu6IXViSQdg)

The only things, that I have found not to work are:
1. The import/export options since they are currently only using an Android specific flutter plugin I think.
2. The tabs are loosing their state when switching back and forth between them (that often triggers an error report too)
    \+ feeds loose state when viewing comments or pictures and returning (doesn't occur on single profile feeds).

Otherwise it already works remarkably well.

Is there anything, that speaks against getting Fritter officially on the Apple Appstore form your side @jonjomckay?

**Comment by bensabo** [comment]
Any update on this at all? @salevdns I've tried using the IPA you provided but was met with numerous errors. 

A shame that this isn't on iOS just yet. I'm getting really tired of the new features Twitter is pushing out and this is the cleanest and nicest app I've seen that's actually tackling said issues as well as privacy. Hope that there's some progress this year on this front.

**Comment by salevdns** [comment]
The version, I posted is now outdated and broken because of API changes, I think.
Unfortunately, I can't build a working IPA from the current source. I don't know why; 
flutter_iconpicker or its implementation in this project seems to be broken but can be circumvented with "--no-tree-shake-icons" build argument, however the resulting IPA just opens a gray screen and does nothing.
I don't know if this is related to "--no-tree-shake-icons" or has a different cause.

**Comment by Loki-L1130** [comment]
Any Updates on this ?

Why not distribute the App as an IPA in the releases or in the AltStore ?

**Comment by Chain-Ksword** [comment]
+1 for an IPA that can be sideloaded if appstore isnt a viable option

**Comment by debpalash** [comment]
My copy is working on real ios device currently, maybe i'll push some change in future for ios @jonjomckay

**Comment by jonjomckay** [comment]
@debpalash yes please, that would be a great help! If you could also detail how you built it, maybe we can get it set up in the GitHub CI too.

**Comment by debpalash** [comment]
> @debpalash yes please, that would be a great help! If you could also detail how you built it, maybe we can get it set up in the GitHub CI too.

I'll when i get some spare time 🤔 Btw Scrolling gesture needs some smoothing, plus you could make the bars on top bottom hide when scrolling up.

**Comment by frorayz** [comment]
> > @debpalash yes please, that would be a great help! If you could also detail how you built it, maybe we can get it set up in the GitHub CI too.
> 
> I'll when i get some spare time 🤔 Btw Scrolling gesture needs some smoothing, plus you could make the bars on top bottom hide when scrolling up.

Could you please share your IPA? I would love to sideload it via AltStore. Thank you!

**Comment by debpalash** [comment]
> Could you please share your IPA? I would love to sideload it via AltStore. Thank you!

I'd send you the ipa, but you should not trust strangers you could be hacked. I'll try to setup ipa build ci and push in this repo. Stay tuned.

**Comment by frorayz** [comment]
> > Could you please share your IPA? I would love to sideload it via AltStore. Thank you!
> 
> I'd send you the ipa, but you should not trust strangers you could be hacked. I'll try to setup ipa build ci and push in this repo. Stay tuned.

Thanks!

**Comment by eusip** [comment]
> > Could you please share your IPA? I would love to sideload it via AltStore. Thank you!
> 
> 
> 
> I'd send you the ipa, but you should not trust strangers you could be hacked. I'll try to setup ipa build ci and push in this repo. Stay tuned.

Any chance I could also get a copy of the .ipa? I would like to try it out with TrollStore.

**Comment by debpalash** [comment]
> Any chance I could also get a copy of the .ipa? I would like to try it out with TrollStore.

@frorayz if you still want 
@eusip share ios version

**Comment by eusip** [comment]
> > Any chance I could also get a copy of the .ipa? I would like to try it out with TrollStore.
> 
> 
> 
> @frorayz if you still want 
> 
> @eusip share ios version
> 
> 
> 
> min ios15.0 
> 
> rename to ipa
> 
> [Fritter.ipa.zip](https://github.com/jonjomckay/fritter/files/9905557/Fritter.ipa.zip)
> 
> 

It works quite well! Thank you!

**Comment by frorayz** [comment]
> > Any chance I could also get a copy of the .ipa? I would like to try it out with TrollStore.
> 
> @frorayz if you still want @eusip share ios version
> 
> min ios15.0 rename to ipa [Fritter.ipa.zip](https://github.com/jonjomckay/fritter/files/9905557/Fritter.ipa.zip)

Thank you!

**Comment by ashutoshsaboo** [comment]
Any news on this? How about releasing on the official ios appstore? Is there a challenge in that? @jonjomckay

**Comment by jonjomckay** [comment]
The biggest challenges are that I don't have a Mac, and the only iOS device I have is my work phone - something I wouldn't be able to use for personal development projects.

Donations, sponsorship, or [funding this issue on IssueHunt](https://issuehunt.io/r/jonjomckay/fritter/issues/55) could help me purchase a cheap Mac to develop and test on!
